pachyderm/templates/pachd/storage-secret.yaml

```diff
@@ -3,45 +3,43 @@ SPDX-FileCopyrightText: Pachyderm, Inc. <info@pachyderm.com>
-  google-cred: {{ .Values.pachd.storage.google.cred | b64enc | quote }}
-  {{- else if eq .Values.pachd.storage.backend "MINIO" }}
-  minio-bucket: {{ .Values.pachd.storage.minio.bucket | toString | b64enc | quote }}
-  minio-endpoint: {{ .Values.pachd.storage.minio.endpoint | toString| b64enc | quote }}
-  minio-id: {{ .Values.pachd.storage.minio.id | toString| b64enc | quote }}
-  minio-secret: {{ .Values.pachd.storage.minio.secret | toString | b64enc | quote }}
-  minio-secure: {{ .Values.pachd.storage.minio.secure | toString | b64enc | quote }}
-  minio-signature: {{ .Values.pachd.storage.minio.signature | toString | b64enc | quote }}
-  {{- else if eq .Values.pachd.storage.backend "AMAZON" }}
-  amazon-bucket: {{ .Values.pachd.storage.amazon.bucket | toString | b64enc | quote }}
-  amazon-distribution: {{ .Values.pachd.storage.amazon.cloudFrontDistribution | toString | b64enc | quote }}
-  {{- if  not .Values.pachd.storage.amazon.iamRole }}
-  amazon-id: {{ .Values.pachd.storage.amazon.id | toString | b64enc | quote }}
-  amazon-region: {{ .Values.pachd.storage.amazon.region | toString | b64enc | quote }}
-  amazon-secret: {{ .Values.pachd.storage.amazon.secret | toString | b64enc | quote }}
-  amazon-token: {{ .Values.pachd.storage.amazon.token | toString | b64enc | quote }}
-  custom-endpoint: {{ .Values.pachd.storage.amazon.customEndpoint | toString | b64enc | quote }}
-  {{- end }}
-  disable-ssl: {{ .Values.pachd.storage.amazon.disableSSL | toString | b64enc | quote }}
-  log-options: {{ .Values.pachd.storage.amazon.logOptions | toString | b64enc | quote }}
-  max-upload-parts: {{ .Values.pachd.storage.amazon.maxUploadParts | toString | b64enc | quote }}
-  no-verify-ssl: {{ not .Values.pachd.storage.amazon.verifySSL | toString | b64enc | quote }}
-  part-size: {{ .Values.pachd.storage.amazon.partSize | b64enc | quote }}
-  retries: {{ .Values.pachd.storage.amazon.retries | toString | b64enc | quote }}
-  reverse: {{ .Values.pachd.storage.amazon.reverse | toString | b64enc | quote }}
-  timeout: {{ .Values.pachd.storage.amazon.timeout | toString | b64enc | quote }}
-  upload-acl: {{ .Values.pachd.storage.amazon.uploadACL | toString | b64enc | quote }}
-  {{- else if eq .Values.pachd.storage.backend "MICROSOFT" }}
-  microsoft-container: {{ .Values.pachd.storage.microsoft.container | toString | b64enc | quote }}
-  microsoft-id: {{ .Values.pachd.storage.microsoft.id | toString | b64enc | quote }}
-  microsoft-secret: {{ .Values.pachd.storage.microsoft.secret | toString | b64enc | quote }}
-  {{- end }}
 kind: Secret
 metadata:
   labels:
     app: pachyderm-storage-secret
     suite: pachyderm
   name: pachyderm-storage-secret
   namespace: {{ .Release.Namespace }}
+data:
+  {{- if eq .Values.pachd.storage.backend "GOOGLE" }}
+  GOOGLE_BUCKET: {{ required "Google bucket Required when storage backend GOOGLE" .Values.pachd.storage.google.bucket | b64enc | quote }}
```

Comment:
"A value for google.bucket is required when the storage backend is GOOGLE."

The helm chart is the first interface with Pachyderm.  We probably want customers to think we can speak English, even if they misconfigure it ;)

Reply:
Agree it's awkward, but what would you suggest?